index.html

```diff
@@ -1343,42 +1355,50 @@
           </p>
           <p>
             When the <code><dfn>close</dfn>()</code> method is called on a
-            <a>PresentationConnection</a>, the user agent MUST run the
-            algorithm to <a data-lt="close-algorithm">close a presentation
-            connection</a> with <a>PresentationConnection</a>, a
-            <code>reason</code> of <code>closed</code>, and an empty
-            <code>message</code>.
+            <a>PresentationConnection</a> <var>S</var>, the <a>user agent</a>
+            MUST <a>start closing the presentation connection</a> <var>S</var>
+            with <code>closed</code> as <code>closeReason</code> and an empty
+            <code>message</code> as <code>closeMessage</code>.
```

Comment:
I think case 3 remains in your latest update: since the connection is transitioned to `closed` in a *task*, the transition won't happen in the same run of the event loop when `close()` is called, and thus the code in the "should never happen" block in my code example above will be executed.

We can't simply drop "queue a task" from the "close a presentation connection" procedure since that procedure is also called from other procedures that run in the background. Adding a step in this portion of the spec to switch the state to `closed` would mostly solve the issue, although the check on the `state` in the close procedure would need to be revisited as well :-(